cloudera/datadog_checks/cloudera/api_client_v7.py

```diff
@@ -0,0 +1,129 @@
+                executor.submit(self._collect_cluster_service_check, cluster, tags)
+                executor.submit(self._collect_hosts, cluster_name)
+
+    @staticmethod
+    def _collect_cluster_tags(cluster, custom_tags):
+        cluster_tags = [f"{cluster_tag.name}:{cluster_tag.value}" for cluster_tag in cluster.tags]
+
+        for custom_tag in custom_tags:
+            cluster_tags.append(custom_tag)
+
+        return cluster_tags
+
+    def _collect_cluster_service_check(self, cluster, tags):
+        cluster_entity_status = ENTITY_STATUS[cluster.entity_status]
+        message = cluster.entity_status if cluster_entity_status != AgentCheck.OK else None
+        self._check.service_check(
+            CLUSTER_HEALTH, cluster_entity_status, tags=tags + [f'cloudera_cluster:{cluster.name}'], message=message
+        )
+
+    def _collect_cluster_metrics(self, cluster_name, tags):
+        metric_names = ','.join(f'last({metric}) AS {metric}' for metric in TIMESERIES_METRICS['cluster'])
+        query = f'SELECT {metric_names} WHERE clusterName="{cluster_name}" AND category=CLUSTER'
+        self._query_time_series(query, tags=tags)
+
+    def _collect_hosts(self, cluster_name):
+        clusters_resource_api = cm_client.ClustersResourceApi(self._api_client)
+        list_hosts_response = clusters_resource_api.list_hosts(cluster_name, view='full')
+        self._log.debug("Cloudera full hosts response:\n%s", list_hosts_response)
+        with ThreadPoolExecutor(max_workers=len(list_hosts_response.items) * 4) as executor:
+            for host in list_hosts_response.items:
+                tags = self._collect_host_tags(host, self._check.config.tags)
+                executor.submit(self._collect_host_metrics, host, tags)
+                executor.submit(self._collect_role_metrics, host, tags)
+                executor.submit(self._collect_disk_metrics, host, tags)
+                executor.submit(self._collect_host_service_check, host, tags)
+
+    @staticmethod
+    def _collect_host_tags(host, custom_tags):
+        tags = [
+            f'cloudera_hostname:{host.hostname}',
+            f'cloudera_rack_id:{host.rack_id}',
+            f'cloudera_cluster:{host.cluster_ref.cluster_name}',
+        ]
+
+        host_tags = host.tags
+        if host_tags:
+            for host_tag in host_tags:
+                tags.append(f"{host_tag.name}:{host_tag.value}")
+
+        for custom_tag in custom_tags:
+            tags.append(custom_tag)
+
+        return tags
+
+    def _collect_host_service_check(self, host, tags):
+        host_entity_status = ENTITY_STATUS[host.entity_status] if host.entity_status else None
+        self._log.debug('Cloudera host_entity_status: %s', host_entity_status)
+        self._check.service_check(HOST_HEALTH, host_entity_status, tags=tags)
+
+    def _collect_host_metrics(self, host, tags):
+        with ThreadPoolExecutor(max_workers=2) as executor:
```

Comment:
Let's add a comment explaining the choice for `max_workers`

Reply:
Will add a comment to mention that for this host, we are executing two tasks in parallel (line 92-93).